config/config.php

```diff
@@ -37,12 +26,27 @@
         } else {
             $proxy = sprintf('http://%s:%s', $proxyHost, $proxyPort);
         }
-        $httpClient = new \Matomo\Dependencies\GoogleAnalyticsImporter\GuzzleHttp\Client([
+        return new \Matomo\Dependencies\GoogleAnalyticsImporter\GuzzleHttp\Client([
             'proxy'      => $proxy,
             'exceptions' => false,
             'base_uri'   => \Matomo\Dependencies\GoogleAnalyticsImporter\Google\Client::API_BASE_PATH
         ]);
-        $googleClient->setHttpClient($httpClient);
+    }
+    return null;
+}, 'GoogleAnalyticsImporter.googleClient' => function (Container $c) {
+    $klass = $c->get('GoogleAnalyticsImporter.googleClientClass');
```

Comment:
@snake14 Won't `$klass` will become null if not proxy ?

Reply:
@AltamashShaikh No, `GoogleAnalyticsImporter.googleClientClass` should always be set regardless of whether proxy is used: https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/932d93534b879441c5cae1e03bab820f2ec1fd56/config/config.php#L17